OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -112,8 +111,7 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 				if (Minefield.Length == 1 && Minefield[0] != self.Location)
 					self.SetTargetLine(Target.FromCell(self.World, Minefield[0]), Color.Red);
 
-				self.CancelActivity();
-				self.QueueActivity(new LayMines(self));
+				self.QueueActivity(order.Queued, new LayMines(self, Minefield));
```

Comment:
Unfortunately this isn't quite enough to fix queued orders here. The trait-level `Minefield` array is set a few lines above, so later queued orders stomp over state required by earlier queued orders.  This will need to move inside the activity, then be pulled out when needed.

Repro case: queue up move, deploy, move, deploy, move, deploy - watch unit run around not knowing what to do.